### ISSUE TITLE
Explore: Query cannot be run when rich history exceeded local storage quota

### ISSUE THREAD

**Issue by ifrost** [grafana/grafana#38814]
**What happened**:

It seems to be rare but may happen in some cases. Explore crashes if richHistory has reached the limit of local storage and the query is run:

https://user-images.githubusercontent.com/745532/131815549-a76ecbb8-d44d-4a59-8f7a-0b4b27d13e82.mov

**What you expected to happen**:

An alert message could be displayed instead indicating that the rich history limit was hit with a possibility to clear the history.

**How to reproduce it (as minimally and precisely as possible)**:

Add this snippet before the history is saved in try/catch block here: https://github.com/grafana/grafana/blob/v8.1.2/public/app/core/utils/richHistory.ts#L77

```typescript
    for (let i = 0; i < 1000; i++) {
      updatedHistory = updatedHistory.concat({
        queries: newQueriesToSave,
        ts,
        datasourceId,
        datasourceName,
        starred,
        comment,
        sessionName,
      });
    }
```

Then run the query a few times.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.1.2
- Data source type & version: Any
- OS Grafana is installed on: Mac
- User OS & Browser: macOS Big Sur
- Grafana plugins: None
- Others: N/A

**Comment by Dieterbe** [comment]
is it normal that even when i wipe all the site data for all sites in firefox, after trying 1 query in explore, i already get this again?

**Comment by ethirolle** [comment]
Note that the when I follow the minified React error link, I see the error is this:

```
Objects are not valid as a React child (found: Error: Could not save item in localStorage: grafana.explore.richHistory. [QuotaExceededError: The quota has been exceeded.]). If you meant to render a collection of children, use an array instead.
```

I find this part interesting:
> If you meant to render a collection of children, use an array instead

Does this suggest a bug that could be fixed by using a different variable structure?

**Comment by ifrost** [comment]
@ethirolle:

> Does this suggest a bug that could be fixed by using a different variable structure?

Yes! The component was passing an Error object instead of a string. Should be fixed by https://github.com/grafana/grafana/pull/39766
